IPython/utils/warn.py

```diff
@@ -10,11 +10,15 @@
 
 import sys
 import warnings
+from IPython.utils.decorators import undoc
 
 warnings.warn("The module IPython.utils.warn is deprecated since IPython 4.0, use the standard warnings module instead", DeprecationWarning)
 
+@undoc
```

Comment:
If the whole of this module should be undocumented, it's easier to add it as a skip pattern in `autogen_api.py` than to mark every function as undoc.